core-api/src/main/java/com/optimizely/ab/internal/CacheWithRemove.java

```diff
@@ -0,0 +1,21 @@
+/**
+ *
+ *    Copyright 2025, Optimizely
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.optimizely.ab.internal;
+
+public interface CacheWithRemove<T> extends Cache<T> {
```

Comment:
java supports default implementation for interface - 
We can remove CacheWithRemove and add remove() to Cache, so backward compatible with existing Cache

    default void remove(String key) {
        // Default implementation does nothing
        // Implementations should override this method to provide actual removal functionality
    }